OpenNefia.Content/UI/Hud/HudMessageWindow.cs

```diff
@@ -0,0 +1,243 @@
+            return cont;
+        }
+
+        private void ReLayoutText()
+        {
+            var lines = new List<MessageContainer>();
+            var currentLine = GetMessageLine(false);
+            Color currentColor = Color.White;
+            var totalWidth = 0;
+            foreach (var message in Messages.Reverse())
+            {
+                foreach(var tag in message.Tags)
+                {
+                    switch (tag)
+                    {
+                        case NewlineMessageTag newLine:
+                            if (currentLine.HasContent)
+                                lines.Add(currentLine);
+                            currentLine = GetMessageLine(true);
+                            totalWidth = 0;
+                            break;
+
+                        case ColorMessageTag colorTag:
+                            currentColor = colorTag.Color;
+                            break;
+
+                        case TextMessageTag textTag:
+                            var sb = new StringBuilder();
+                            var words = UiHelpers.SplitString(textTag.Message, Loc.Language);
+                            foreach(var word in words)
+                            {
+                                var wordWidth = UiFonts.MessageText.LoveFont.GetWidth(word);
+                                if (totalWidth + wordWidth > Width)
+                                {
+                                    currentLine.AddElement(new MessageText(sb.ToString().TrimStart()) { Color = currentColor });
+                                    lines.Add(currentLine);
+                                    currentLine = GetMessageLine(false);
+                                    totalWidth = 0;
+                                    sb.Clear();
+                                }
+                                sb.Append(word);
+                                totalWidth += wordWidth;
+                            }
+                            currentLine.AddElement(new MessageText($"{sb.ToString().TrimStart()} ") { Color = currentColor });
+                            break;
+
+                        case IconMessageTag icon:
+                            //TODO display icons
+                            break;
+                    }
+                }
+            }
+            lines.Add(currentLine);
+            MessageBoxContainer.Clear();
+            BacklogContainer.Clear();
+
+            var emptyLine = new MessageContainer();
+            emptyLine.AddElement(new UiText(UiFonts.MessageText));
+
+            var lineIndex = 0;
+            var mesBoxLines = Enumerable.Empty<MessageContainer>()
+                .Concat(Enumerable.Repeat(emptyLine, Math.Max(0, MessageBoxLines - lines.Count)))
+                .Concat(lines.Skip(Math.Max(0, lines.Count - MessageBoxLines))); 
+            var backlogLines = lines.Count > MessageBoxLines
+                ? Enumerable.Empty<MessageContainer>()
```

Comment:
You can just do the `Enumerable.Repeat` here instead of calling `Enumerable.Empty<>()` first.

Reply:
changed in 9eb449f